### DISCUSSION TITLE
Setting reporting interval?

### DISCUSSION THREAD

**Discussion by mattmoo** [Koenkk/zigbee2mqtt#13279] [Devices]
I'm currently experimenting with Zigbee2MQTT to see if I can move my Zigbee devices to a separate Docker, like I did with Zwave.

I have a few [TuYa WSD500A](https://www.zigbee2mqtt.io/devices/WSD500A.html#tuya-wsd500a), and they are reporting fine, but one thing that seems to be escaping me is how to set the reproting interval.

It seems that no matter what I try and set in the 'Reporting' tab, it reports that the bind failed, as below. Is there something that I am missing?

![image](https://user-images.githubusercontent.com/20387030/180675098-be58f033-7868-46dc-ad62-572369bc80dc.png)

**Comment by sjorge** [comment]
I'm not sure about this particular one but the TuYa test sensor I have does not implement zigbee reporting and sends the information in the TuYa format piggy backing over the zigbee mesh. I'm guessing this is the same for this one.

**Comment by mattmoo** [comment]
After a bit of googling, I understand... some of those words. Does zigbee2mqtt have a means to effectively increase the sample rate of the TuYa format yet?

**Comment by 0crap** [comment]
Subscribed.
Also not able to change the reporting interval on my TuYa WSD500A devices.
It seems to report now every 30 minutes. Which is OK, but I prefer to have it a little faster.

**Comment by EvertDekker** [comment]
Don't forget to wake up the device before sending the report interval changes.
For this device I think pressing short the reset button.

**Reply by GrauBlitz** [reply]
This worked for me too! (Atleast setting it, let's see if the reports come in)

**Reply by lennon101** [reply]
how did you go with this? were you able to change the rep interval?

**Comment by 0crap** [comment]
Hero! That indeed helps and gets the mqtt message accepted!
Any idea what unit I need to enter into the 'min rep interval' box?
mS, sec, min, hour?
It seems to have a will on it's own.

**Reply by lux73** [reply]
afaik all reporting Entrys are in Seconds

**Reply by 0crap** [reply]
Thx!

**Comment by OneCreek** [comment]
Hi togehter, 

any idea how to update in the interval with https://github.com/ioBroker/ioBroker.zigbee/issues/1649?

Thanks!

**Comment by alexsannikov** [comment]
Guys, what exactly I should put into each field, if I chose 1-5 mins interval? Which cluster and attribute? Can somebody give any clue or example? Thanks

**Reply by OlafKocanda** [reply]
depends on what you want to report. is it just the OnOff status? then you can enter the following:
![grafik](https://user-images.githubusercontent.com/58265203/213421171-da0fffa4-5205-4ed7-a5f9-5ec153065e6d.png)

**Comment by OlafKocanda** [comment]
I was wondering whether it is possible to change the OnOff attribute for several devices at once? E.g. setting a standard reporting interval for all devices for this attribute. 
I find it pretty disturbing going through all my lamps one after another in the reporting tab

**Reply by sjorge** [reply]
Just wait until one of your models triggers and update and you have to do it again -_-

I just run this after every z2m update:
```
#!/usr/bin/bash

## Bulbs
## <type=c,ws,cws> <brand=g,h> <name>
# cws h light/bathroom/mood/bulb
# cws h light/office/mood/bulb
# cws h light/livingroom/mood/bulb1
# cws h light/livingroom/mood/bulb2
# ws g light/balcony/main/bulb
# cws h light/bedroom/main/bulb1
# cws h light/bedroom/main/bulb2
# cws h light/bedroom/main/bulb3
# cws h light/bedroom/main/bulb4
# ws h light/entrance/main/bulb
# ws h light/kitchen/counter/bulb1
# ws h light/kitchen/counter/bulb2
# ws h light/kitchen/counter/bulb3
# ws h light/kitchen/table/bulb1
# ws h light/kitchen/table/bulb2
# ws h light/kitchen/table/bulb3
# ws h light/kitchen/table/bulb4
# ws h light/livingroom/main/bulb1
# ws h light/livingroom/main/bulb2
# ws h light/livingroom/main/bulb3
# ws h light/livingroom/main/bulb4
# cws h light/hallway/main/bulb
# ws g light/pantry/main/bulb1
# ws g light/pantry/main/bulb2
# ws g light/bathroom/main/panel1
# ws g light/bathroom/main/panel2
# ws g light/office/main/panel1
# ws g light/office/main/panel2
# ws g light/office/main/panel3
# ws g light/office/main/panel4
# ws g light/office/desk/bulb

## loop and setup
grep -E '^# (cws|ws|c) ' $0 | \
while read bulb; do
        btype=$(awk '{ print $2 }' <<< $bulb)
        bbrand=$(awk '{ print $3 }' <<< $bulb)
        bname=$(awk '{ print $4 }' <<< $bulb)

        echo -ne "[....] Setting up ${bname} as type ${btype} ..."
        # NOTE: power on and color change when off behavior
        mosquitto_pub -t zigbee2mqtt/${bname}/set/power_on_behavior -m 'off'

        # NOTE: state and level reporting
        mosquitto_pub -t zigbee2mqtt/bridge/request/device/configure_reporting \
                -m '{"id":"'${bname}'","cluster":"genOnOff","attribute":"onOff","minimum_report_interval":0,"maximum_report_interval":3600,"reportable_change":0}'
        mosquitto_pub -t zigbee2mqtt/bridge/request/device/configure_reporting \
                -m '{"id":"'${bname}'","cluster":"genLevelCtrl","attribute":"currentLevel","minimum_report_interval":3,"maximum_report_interval":3600,"reportable_change":1}'

        sleep 1

        # NOTE: color reporting
        case $btype in
                cws|ws|c)
                        if [ "$bbrand" != "h" ]; then
                                mosquitto_pub -t zigbee2mqtt/bridge/request/device/configure_reporting \
                                        -m '{"id":"'${bname}'","cluster":"lightingColorCtrl","attribute":"colorMode","minimum_report_interval":3,"maximum_report_interval":3600,"reportable_change":1}'
                        fi
                ;;
        esac
        case $btype in
                cws|ws)
                        mosquitto_pub -t zigbee2mqtt/bridge/request/device/configure_reporting \
                                -m '{"id":"'${bname}'","cluster":"lightingColorCtrl","attribute":"colorTemperature","minimum_report_interval":3,"maximum_report_interval":3600,"reportable_change":1}'
                        mosquitto_pub -t zigbee2mqtt/bridge/request/device/configure_reporting \
                                -m '{"id":"'${bname}'","cluster":"lightingColorCtrl","attribute":"currentX","minimum_report_interval":3,"maximum_report_interval":3600,"reportable_change":1}'
                        mosquitto_pub -t zigbee2mqtt/bridge/request/device/configure_reporting \
                                -m '{"id":"'${bname}'","cluster":"lightingColorCtrl","attribute":"currentY","minimum_report_interval":3,"maximum_report_interval":3600,"reportable_change":1}'
                ;;
        esac
        case $btype in
                cws|c)
                        mosquitto_pub -t zigbee2mqtt/bridge/request/device/configure_reporting \
                                -m '{"id":"'${bname}'","cluster":"lightingColorCtrl","attribute":"currentHue","minimum_report_interval":3,"maximum_report_interval":3600,"reportable_change":1}'
                        mosquitto_pub -t zigbee2mqtt/bridge/request/device/configure_reporting \
                                -m '{"id":"'${bname}'","cluster":"lightingColorCtrl","attribute":"currentSaturation","minimum_report_interval":3,"maximum_report_interval":3600,"reportable_change":1}'
                ;;
        esac
        sleep 1

        # NOTE: color options
        case $btype in
                cws|ws|c)
                        mosquitto_pub -t zigbee2mqtt/${bname}/set/color_options -m '{"execute_if_off": true}'
                ;;
        esac
        # XXX: seems to not be support by hue, new tradfri stuff
        #mosquitto_pub -t zigbee2mqtt/${bname}/set/level_options -m '{"execute_if_off": true}'
        sleep 5
        echo -e "\r[DONE"
done
```

**Reply by OlafKocanda** [reply]
Wait, do I understand you right, that after every update of z2m we have to set the reporting interval again?

**Reply by sjorge** [reply]
Not after every update, but I do it anyway, You have to reset them anytime the following happens:
- remote bind/unbind from the light (or group the light is part of)
- `configure()` for the light gets updated (if the configure does something with the reporting, a lot of them do)
- (not 100% sure on this one) the light gets added/removed from a group that has a remote bound.

**Comment by gwww** [comment]
I did something similar, but different than the above script. Part of what I am doing is disabling Home Assistant entities as well as some config of reporting. I was targetting the new Inovelli switch that creates soooooooo many HA entities and has "agressive" reporting.

https://gist.github.com/gwww/1c959a00ec2bcf59f7041fe731b33a3c

Check out the two files and all the comments on how to install and run it.

**Comment by lennon101** [comment]
Just to add further information to this thread, I believe the "update interval" is not time based but instead is controlled by the device itself. When no change is measured, no data is sent. When a change in temp/rh is detected, it sends a measurement. 

From Tuya:
> Upon successful net pairing, the sensor reports its status. It then takes samples of the environment every five minutes. When the ambient temperature exceeds the preset value of the sensor by 0.6°C or the humidity exceeds the preset value by 6%, the sensor will send an alarm to TuyaSmart. Regardless of whether the temperature or humidity exceeds the preset thresholds, data will be synchronized to TuyaSmart every 30 minutes. 

More info here:

https://support.tuya.com/en/help/_detail/K9hw4fohmnwuw

**Reply by zapdogster** [reply]
For me this worked. I had to click on "Apply" first in HomeSeer, then press the WDS500A reset button momentarily (the opposite doesn't work). It now reports every 10 minutes instead of 30 minutes. Resulting HS screen:

![2024-04-17 14_46_17-Reporting_Interval docx  Compatibility Mode  - Word](https://github.com/Koenkk/zigbee2mqtt/assets/6640699/61c0d768-cffc-4b81-b51c-248ccafb74d5)

**Reply by lennon101** [reply]
By "home seer" I'm guessing you meant "Zigbee2mqtt" 😛 that is a screenshot of the Zigbee2mqtt dashboard

**Reply by zapdogster** [reply]
Yes - that is the "view" I get using the zigbee2mqyt plugin in Homeseer. No idea how it looks in other platforms, if there is such. :)

**Reply by lennon101** [reply]
haha yes it looks the same in all platforms (it is just the zigbee2mqtt dashboard being rendered inside either Home Seer or Home Assistant etc). Here is a screenshot of what it looks like inside Home Assistant. As you can see, it is the same. 

<img width="1512" alt="Screenshot 2024-04-18 at 5 58 57 PM" src="https://github.com/Koenkk/zigbee2mqtt/assets/1709944/9ea33a88-a371-471b-81f6-c4fdaea3b335">

**Comment by wise-rice** [comment]
#21286
